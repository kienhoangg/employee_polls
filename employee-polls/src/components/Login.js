import React from "react";
import { useState } from "react";
import { connect } from "react-redux";
import { authenUser } from "../actions/shared";

const Login = (props) => {
  const [user, setUser] = useState("");
  const [password, setPassword] = useState("");
  const handleUser = (e) => {
    setUser(e.target.value);
  };
  const handlePassword = (e) => {
    setPassword(e.target.value);
  };
  const [success, setSuccess] = useState(false);
  const [error, setError] = useState(false);
  const handleLogin = (e) => {
    e.preventDefault();
    if (user === "" || password === "") {
      setSuccess(false);
      setError(true);
      return;
    }
    setSuccess(true);
    setError(false);
    props.dispatch(authenUser(user, password));
    setUser("");
    setPassword("");
  };
  return (
    <form className="new-question" onSubmit={handleLogin}>
      {success && (
        <h1 className={"Success"} data-testid="success-header">
          Name Submitted!
        </h1>
      )}
      {error && (
        <h1 className={"Error"} data-testid="error-header">
          Please enter user and password.
        </h1>
      )}
      <h3 className="center">Login</h3>
      <h4 className="center">User</h4>
      <input
        type="text"
        name="Username"
        data-testid="user"
        className="btnOptionOne"
        placeholder="User"
        value={user}
        onChange={handleUser}
      />
      <h4 className="center">Password</h4>
      <input
        type="password"
        name="Password"
        data-testid="password"
        className="btnOptionOne"
        placeholder="Password"
        value={password}
        onChange={handlePassword}
      />
      <button
        data-testid="submit-button"
        className="btn"
        type="submit"
        // disabled={user === "" || password === ""}
      >
        Submit
      </button>
    </form>
  );
};

export default connect()(Login);
