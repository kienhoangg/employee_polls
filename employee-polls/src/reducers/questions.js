import {
  GET_QUESTIONS,
  ANSWER_QUESTION,
  ADD_NEW_QUESTION,
} from "../actions/questions";

export default function questions(state = {}, action) {
  switch (action.type) {
    case GET_QUESTIONS:
      return {
        ...state,
        ...action.questions,
      };
    case ANSWER_QUESTION:
      return {
        ...state,
      };
    case ADD_NEW_QUESTION:
      return {
        ...state,
        ...action.questions,
      };

    default:
      return state;
  }
}
