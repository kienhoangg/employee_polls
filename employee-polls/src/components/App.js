import React from "react";
import { useEffect, Fragment } from "react";
import { handleInitialData } from "../actions/shared";
import LoadingBar from "react-redux-loading-bar";
import { connect } from "react-redux";
import Login from "./Login";
import Dashboard from "./Dashboard";
import { Route, Routes } from "react-router-dom";
import Navbar from "./Navbar";
import LeaderBoard from "./LeaderBoard";
import Poll from "./Poll";
import New from "./New";
const App = (props) => {
  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  return (
    <Fragment>
      <LoadingBar />
      <h3 className="center">Employee Polls</h3>

      {!props.authedUser ? (
        <Login />
      ) : (
        <div>
          {" "}
          <Navbar />
          <Routes>
            <Route path="/" exact element={<Dashboard />} />
            <Route path="/leaderboard" exact element={<LeaderBoard />} />
            <Route path="/question/:id" element={<Poll />} />
            <Route path="/new" exact element={<New />} />
          </Routes>
        </div>
      )}
    </Fragment>
  );
};
const mapStateToProps = ({ authedUser }) => ({
  loading: authedUser === null,
  authedUser,
});
export default connect(mapStateToProps)(App);
