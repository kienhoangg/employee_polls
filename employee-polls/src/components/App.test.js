import { _saveQuestionAnswer, _saveQuestion } from "../utils/_DATA";
import { fireEvent, render } from "@testing-library/react";
import * as React from "react";
import { Provider } from "react-redux";
import reducers from "../reducers";
import middlewares from "../middlewares";
import { createStore } from "redux";
import { MemoryRouter } from "react-router-dom";
import App from "./App";
import Login from "./Login";
describe("test _saveQuestionAnswer", () => {
  it("will return true if success answer the question", async () => {
    const usersState = {
      sarahedo: {
        id: "sarahedo",
        password: "password123",
        name: "Sarah Edo",
        avatarURL: "https://randomuser.me/api/portraits/women/13.jpg",
        answers: {
          "8xf0y6ziyjabvozdd253nd": "optionOne",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
          am8ehyc8byjqgar0jgpub9: "optionTwo",
          loxhs1bqm25b708cmbf3g: "optionTwo",
        },
        questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
      },
      tylermcginnis: {
        id: "tylermcginnis",
        password: "abc321",
        name: "Tyler McGinnis",
        avatarURL: "https://randomuser.me/api/portraits/men/79.jpg",
        answers: {
          vthrdm985a262al8qx3do: "optionOne",
          xj352vofupe1dqz9emx13r: "optionTwo",
        },
        questions: ["loxhs1bqm25b708cmbf3g", "vthrdm985a262al8qx3do"],
      },
      mtsamis: {
        id: "mtsamis",
        password: "xyz123",
        name: "Mike Tsamis",
        avatarURL: "https://randomuser.me/api/portraits/women/26.jpg",
        answers: {
          xj352vofupe1dqz9emx13r: "optionOne",
          vthrdm985a262al8qx3do: "optionTwo",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
        },
        questions: ["6ni6ok3ym7mf1p33lnez", "xj352vofupe1dqz9emx13r"],
      },
      zoshikanlu: {
        id: "zoshikanlu",
        password: "pass246",
        name: "Zenobia Oshikanlu",
        avatarURL: "https://randomuser.me/api/portraits/women/54.jpg",
        answers: {
          xj352vofupe1dqz9emx13r: "optionOne",
        },
        questions: [],
      },
    };

    var questionsState = {
      "8xf0y6ziyjabvozdd253nd": {
        id: "8xf0y6ziyjabvozdd253nd",
        author: "sarahedo",
        timestamp: 1467166872634,
        optionOne: {
          votes: ["sarahedo"],
          text: "Build our new application with Javascript",
        },
        optionTwo: {
          votes: [],
          text: "Build our new application with Typescript",
        },
      },
      "6ni6ok3ym7mf1p33lnez": {
        id: "6ni6ok3ym7mf1p33lnez",
        author: "mtsamis",
        timestamp: 1468479767190,
        optionOne: {
          votes: [],
          text: "hire more frontend developers",
        },
        optionTwo: {
          votes: ["mtsamis", "sarahedo"],
          text: "hire more backend developers",
        },
      },
      am8ehyc8byjqgar0jgpub9: {
        id: "am8ehyc8byjqgar0jgpub9",
        author: "sarahedo",
        timestamp: 1488579767190,
        optionOne: {
          votes: [],
          text: "conduct a release retrospective 1 week after a release",
        },
        optionTwo: {
          votes: ["sarahedo"],
          text: "conduct release retrospectives quarterly",
        },
      },
      loxhs1bqm25b708cmbf3g: {
        id: "loxhs1bqm25b708cmbf3g",
        author: "tylermcginnis",
        timestamp: 1482579767190,
        optionOne: {
          votes: [],
          text: "have code reviews conducted by peers",
        },
        optionTwo: {
          votes: ["sarahedo"],
          text: "have code reviews conducted by managers",
        },
      },
      vthrdm985a262al8qx3do: {
        id: "vthrdm985a262al8qx3do",
        author: "tylermcginnis",
        timestamp: 1489579767190,
        optionOne: {
          votes: ["tylermcginnis"],
          text: "take a course on ReactJS",
        },
        optionTwo: {
          votes: ["mtsamis"],
          text: "take a course on unit testing with Jest",
        },
      },
      xj352vofupe1dqz9emx13r: {
        id: "xj352vofupe1dqz9emx13r",
        author: "mtsamis",
        timestamp: 1493579767190,
        optionOne: {
          votes: ["mtsamis", "zoshikanlu"],
          text: "deploy to production once every two weeks",
        },
        optionTwo: {
          votes: ["tylermcginnis"],
          text: "deploy to production once every month",
        },
      },
    };

    var questionsResult = {
      "8xf0y6ziyjabvozdd253nd": {
        id: "8xf0y6ziyjabvozdd253nd",
        author: "sarahedo",
        timestamp: 1467166872634,
        optionOne: {
          votes: ["sarahedo"],
          text: "Build our new application with Javascript",
        },
        optionTwo: {
          votes: [],
          text: "Build our new application with Typescript",
        },
      },
      "6ni6ok3ym7mf1p33lnez": {
        id: "6ni6ok3ym7mf1p33lnez",
        author: "mtsamis",
        timestamp: 1468479767190,
        optionOne: {
          votes: [],
          text: "hire more frontend developers",
        },
        optionTwo: {
          votes: ["mtsamis", "sarahedo"],
          text: "hire more backend developers",
        },
      },
      am8ehyc8byjqgar0jgpub9: {
        id: "am8ehyc8byjqgar0jgpub9",
        author: "sarahedo",
        timestamp: 1488579767190,
        optionOne: {
          votes: [],
          text: "conduct a release retrospective 1 week after a release",
        },
        optionTwo: {
          votes: ["sarahedo"],
          text: "conduct release retrospectives quarterly",
        },
      },
      loxhs1bqm25b708cmbf3g: {
        id: "loxhs1bqm25b708cmbf3g",
        author: "tylermcginnis",
        timestamp: 1482579767190,
        optionOne: {
          votes: [],
          text: "have code reviews conducted by peers",
        },
        optionTwo: {
          votes: ["sarahedo"],
          text: "have code reviews conducted by managers",
        },
      },
      vthrdm985a262al8qx3do: {
        id: "vthrdm985a262al8qx3do",
        author: "tylermcginnis",
        timestamp: 1489579767190,
        optionOne: {
          votes: ["tylermcginnis"],
          text: "take a course on ReactJS",
        },
        optionTwo: {
          votes: ["mtsamis"],
          text: "take a course on unit testing with Jest",
        },
      },
      xj352vofupe1dqz9emx13r: {
        id: "xj352vofupe1dqz9emx13r",
        author: "mtsamis",
        timestamp: 1493579767190,
        optionOne: {
          votes: ["mtsamis", "zoshikanlu", "sarahedo"],
          text: "deploy to production once every two weeks",
        },
        optionTwo: {
          votes: ["tylermcginnis"],
          text: "deploy to production once every month",
        },
      },
    };
    var usersResult = {
      sarahedo: {
        id: "sarahedo",
        password: "password123",
        name: "Sarah Edo",
        avatarURL: "https://randomuser.me/api/portraits/women/13.jpg",
        answers: {
          "8xf0y6ziyjabvozdd253nd": "optionOne",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
          am8ehyc8byjqgar0jgpub9: "optionTwo",
          loxhs1bqm25b708cmbf3g: "optionTwo",
          xj352vofupe1dqz9emx13r: "optionOne",
        },
        questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
      },
      tylermcginnis: {
        id: "tylermcginnis",
        password: "abc321",
        name: "Tyler McGinnis",
        avatarURL: "https://randomuser.me/api/portraits/men/79.jpg",
        answers: {
          vthrdm985a262al8qx3do: "optionOne",
          xj352vofupe1dqz9emx13r: "optionTwo",
        },
        questions: ["loxhs1bqm25b708cmbf3g", "vthrdm985a262al8qx3do"],
      },
      mtsamis: {
        id: "mtsamis",
        password: "xyz123",
        name: "Mike Tsamis",
        avatarURL: "https://randomuser.me/api/portraits/women/26.jpg",
        answers: {
          xj352vofupe1dqz9emx13r: "optionOne",
          vthrdm985a262al8qx3do: "optionTwo",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
        },
        questions: ["6ni6ok3ym7mf1p33lnez", "xj352vofupe1dqz9emx13r"],
      },
      zoshikanlu: {
        id: "zoshikanlu",
        password: "pass246",
        name: "Zenobia Oshikanlu",
        avatarURL: "https://randomuser.me/api/portraits/women/54.jpg",
        answers: {
          xj352vofupe1dqz9emx13r: "optionOne",
        },
        questions: [],
      },
    };
    var authedUser = "sarahedo";
    var qid = "xj352vofupe1dqz9emx13r";
    var answer = "optionOne";
    var result = await _saveQuestionAnswer({
      authedUser,
      qid,
      answer,
      usersState,
      questionsState,
    });
    expect(result).toEqual({
      users: usersResult,
      questions: questionsResult,
    });
  });

  it("will return an error if the params are not found", async () => {
    const usersState = {
      sarahedo: {
        id: "sarahedo",
        password: "password123",
        name: "Sarah Edo",
        avatarURL: "https://randomuser.me/api/portraits/women/13.jpg",
        answers: {
          "8xf0y6ziyjabvozdd253nd": "optionOne",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
          am8ehyc8byjqgar0jgpub9: "optionTwo",
          loxhs1bqm25b708cmbf3g: "optionTwo",
        },
        questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
      },
      tylermcginnis: {
        id: "tylermcginnis",
        password: "abc321",
        name: "Tyler McGinnis",
        avatarURL: "https://randomuser.me/api/portraits/men/79.jpg",
        answers: {
          vthrdm985a262al8qx3do: "optionOne",
          xj352vofupe1dqz9emx13r: "optionTwo",
        },
        questions: ["loxhs1bqm25b708cmbf3g", "vthrdm985a262al8qx3do"],
      },
      mtsamis: {
        id: "mtsamis",
        password: "xyz123",
        name: "Mike Tsamis",
        avatarURL: "https://randomuser.me/api/portraits/women/26.jpg",
        answers: {
          xj352vofupe1dqz9emx13r: "optionOne",
          vthrdm985a262al8qx3do: "optionTwo",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
        },
        questions: ["6ni6ok3ym7mf1p33lnez", "xj352vofupe1dqz9emx13r"],
      },
      zoshikanlu: {
        id: "zoshikanlu",
        password: "pass246",
        name: "Zenobia Oshikanlu",
        avatarURL: "https://randomuser.me/api/portraits/women/54.jpg",
        answers: {
          xj352vofupe1dqz9emx13r: "optionOne",
        },
        questions: [],
      },
    };

    var questionsState = {
      "8xf0y6ziyjabvozdd253nd": {
        id: "8xf0y6ziyjabvozdd253nd",
        author: "sarahedo",
        timestamp: 1467166872634,
        optionOne: {
          votes: ["sarahedo"],
          text: "Build our new application with Javascript",
        },
        optionTwo: {
          votes: [],
          text: "Build our new application with Typescript",
        },
      },
      "6ni6ok3ym7mf1p33lnez": {
        id: "6ni6ok3ym7mf1p33lnez",
        author: "mtsamis",
        timestamp: 1468479767190,
        optionOne: {
          votes: [],
          text: "hire more frontend developers",
        },
        optionTwo: {
          votes: ["mtsamis", "sarahedo"],
          text: "hire more backend developers",
        },
      },
      am8ehyc8byjqgar0jgpub9: {
        id: "am8ehyc8byjqgar0jgpub9",
        author: "sarahedo",
        timestamp: 1488579767190,
        optionOne: {
          votes: [],
          text: "conduct a release retrospective 1 week after a release",
        },
        optionTwo: {
          votes: ["sarahedo"],
          text: "conduct release retrospectives quarterly",
        },
      },
      loxhs1bqm25b708cmbf3g: {
        id: "loxhs1bqm25b708cmbf3g",
        author: "tylermcginnis",
        timestamp: 1482579767190,
        optionOne: {
          votes: [],
          text: "have code reviews conducted by peers",
        },
        optionTwo: {
          votes: ["sarahedo"],
          text: "have code reviews conducted by managers",
        },
      },
      vthrdm985a262al8qx3do: {
        id: "vthrdm985a262al8qx3do",
        author: "tylermcginnis",
        timestamp: 1489579767190,
        optionOne: {
          votes: ["tylermcginnis"],
          text: "take a course on ReactJS",
        },
        optionTwo: {
          votes: ["mtsamis"],
          text: "take a course on unit testing with Jest",
        },
      },
      xj352vofupe1dqz9emx13r: {
        id: "xj352vofupe1dqz9emx13r",
        author: "mtsamis",
        timestamp: 1493579767190,
        optionOne: {
          votes: ["mtsamis", "zoshikanlu"],
          text: "deploy to production once every two weeks",
        },
        optionTwo: {
          votes: ["tylermcginnis"],
          text: "deploy to production once every month",
        },
      },
    };

    var authedUser = "sarahedo";
    var qid = "xj352vofupe1dqz9emx13r";
    var answer = "optionOne";

    await expect(
      _saveQuestionAnswer({
        authedUser,
        qids: qid,
        answer,
        usersState,
        questionsState,
      })
    ).rejects.toEqual("Please provide authedUser, qid, and answer");
  });
});

describe("test _saveQuestion", () => {
  it("will return true if success add new the question", async () => {
    var optionTwoText = "Click this for option 2";
    var optionOneText = "Click this for option 1";
    var author = "sarahedo";
    var question = {
      optionOneText,
      optionTwoText,
      author,
    };
    var result = await _saveQuestion(question);
    var questionResult = {
      id: "9bx6zkznrai3q4df9ovul",
      timestamp: 1687020328516,
      author: "sarahedo",
      optionOne: { votes: [], text: "Click this for option 1" },
      optionTwo: { votes: [], text: "Click this for option 2" },
    };
    expect(result.author).toEqual(questionResult.author);
    expect(result.optionOne).toEqual(questionResult.optionOne);
    expect(result.optionTwo).toEqual(questionResult.optionTwo);
  });
  it("will return an error if params are not found", async () => {
    await expect(
      _saveQuestion({
        optionOneTexts: "optionOneTexts",
        optionTwoTexts: "optionTwoTexts",
      })
    ).rejects.toEqual(
      "Please provide optionOneText, optionTwoText, and author"
    );
  });

  it("will return an error if option text one not found", async () => {
    await expect(
      _saveQuestion({
        optionTwoTexts: "optionTwoTexts",
      })
    ).rejects.toEqual(
      "Please provide optionOneText, optionTwoText, and author"
    );
  });
});

describe("test snapshot", () => {
  it("will match snapshot", () => {
    const store = createStore(reducers, middlewares);
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});

describe("test firechange", () => {
  it("will match if not input both user and password ", () => {
    const store = createStore(reducers, middlewares);
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );
    var submitButton = component.getByTestId("submit-button");
    fireEvent.click(submitButton);
    expect(component.queryByTestId("error-header")).toBeInTheDocument();
    expect(component.queryByTestId("success-header")).not.toBeInTheDocument();
  });

  it("will match if just input passwsord ", () => {
    const store = createStore(reducers, middlewares);
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );
    var input = component.getByTestId("password");
    fireEvent.change(input, { target: { value: "password123" } });
    var submitButton = component.getByTestId("submit-button");
    fireEvent.click(submitButton);
    expect(component.queryByTestId("error-header")).toBeInTheDocument();
    expect(component.queryByTestId("success-header")).not.toBeInTheDocument();
  });

  it("will match if just input user ", () => {
    const store = createStore(reducers, middlewares);
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );
    var input = component.getByTestId("user");
    fireEvent.change(input, { target: { value: "sarahedo" } });
    var submitButton = component.getByTestId("submit-button");
    fireEvent.click(submitButton);
    expect(component.queryByTestId("error-header")).toBeInTheDocument();
    expect(component.queryByTestId("success-header")).not.toBeInTheDocument();
  });

  it("will match if input both user and password ", () => {
    const store = createStore(reducers, middlewares);
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );
    var inputUser = component.getByTestId("user");
    fireEvent.change(inputUser, { target: { value: "sarahedo" } });

    var inputPassword = component.getByTestId("password");
    fireEvent.change(inputPassword, { target: { value: "password123" } });
    var submitButton = component.getByTestId("submit-button");
    fireEvent.click(submitButton);
    expect(component.queryByTestId("error-header")).not.toBeInTheDocument();
    expect(component.queryByTestId("success-header")).toBeInTheDocument();
  });
});
