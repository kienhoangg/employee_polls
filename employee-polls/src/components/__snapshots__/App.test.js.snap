// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="direction: ltr;"
      >
        <div
          class=""
          style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
        />
        <div
          style="display: table; clear: both;"
        />
      </div>
      <h3
        class="center"
      >
        Employee Polls
      </h3>
      <form
        class="new-question"
      >
        <h3
          class="center"
        >
          Login
        </h3>
        <h4
          class="center"
        >
          User
        </h4>
        <input
          class="btnOptionOne"
          data-testid="user"
          name="Username"
          placeholder="User"
          type="text"
          value=""
        />
        <h4
          class="center"
        >
          Password
        </h4>
        <input
          class="btnOptionOne"
          data-testid="password"
          name="Password"
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          class="btn"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <div
      style="direction: ltr;"
    >
      <div
        class=""
        style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
      />
      <div
        style="display: table; clear: both;"
      />
    </div>
    <h3
      class="center"
    >
      Employee Polls
    </h3>
    <form
      class="new-question"
    >
      <h3
        class="center"
      >
        Login
      </h3>
      <h4
        class="center"
      >
        User
      </h4>
      <input
        class="btnOptionOne"
        data-testid="user"
        name="Username"
        placeholder="User"
        type="text"
        value=""
      />
      <h4
        class="center"
      >
        Password
      </h4>
      <input
        class="btnOptionOne"
        data-testid="password"
        name="Password"
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        class="btn"
        data-testid="submit-button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
